### PR TITLE
[JIT] Move Shape Function definition to python

### PR DESCRIPTION
**Pull request by eellison**
Moves jit shape function registration to python. Like jit decompositions, a script must be run after adding new definitions which serializes them in a c++ file. 

This was a request so that torch-mlir could define functions in python and upstream their shape functions. cc @silvasean  @makslevental

### CODE REVIEW

**Comment by makslevental** [comment]
@eellison Good solution to the problem of getting access in both cpp and python. Btw couldn't this be a part of `codegen.py` (or something like that)?

**Comment by eellison** [comment]
@makslevental couldn't be part of codegen because there's a dependency on pytorch already beingg buit

**Comment by silvasean** [comment]
Wonderful. Can't wait for this to land and to plug this in!

File: tools/codegen/shape_functions/gen_jit_shape_functions.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+// clang-format on
+
+} // namespace jit
+} // namespace torch
+"""
+
+
+DECOMPOSITION_UTIL_FILE_NAME = "serialized_shape_function_registry.cpp"
+
+def gen_serialized_decompisitions() -> str:
```
Comment (davidberard98) [inline tools/codegen/shape_functions/gen_jit_shape_functions.py:56]: nit - decompositions

**Review by davidberard98 — APPROVED** [review]
looks good!

**Pull request closed without merging**